hashicorp/boundary-ui
# chore: handle nested data from static json credentials

# Description
This PR handles nested data from JSON type credentials same as Vault 
https://hashicorp.atlassian.net/browse/ICU-15854



## Screenshots (if appropriate)
before:
<img width="662" alt="Screenshot 2024-12-04 at 12 45 41 PM" src="https://github.com/user-attachments/assets/ef38f0d2-9bc7-4013-8fe1-374f7a0cd837">


after:
using mocks:
<img width="750" alt="Screenshot 2024-12-05 at 10 29 30 AM" src="https://github.com/user-attachments/assets/01de506b-7523-40ee-8726-7b18c08022ff">

using boundary controller:
<img width="1081" alt="Screenshot 2024-12-05 at 10 04 58 AM" src="https://github.com/user-attachments/assets/560af52a-18fd-4cd1-afe4-757a2edb0ffd">


## How to Test

- create a static JSON credential with a nested object
- add the credential as a brokered credential to the target
- select the above target ID in the desktop app and click connect
- all the fields of the credential should show in a key/value format
## Checklist
